r/tests/testthat/test-compute-sort.R

```diff
@@ -0,0 +1,143 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+context("compute: sorting")
+
+library(dplyr)
+
+tbl <- example_data_for_sorting
+
+test_that("sort(Scalar) is identity function", {
+  expect_identical(
+    as.vector(sort(Scalar$create(42L))),
+    42L
+  )
+  expect_identical(
+    as.vector(sort(Scalar$create("foo"))),
+    "foo"
+  )
+})
+
+test_that("Array$SortIndices()", {
+  expect_equal(
+    Array$create(tbl$int)$SortIndices(),
+    Array$create(0L:9L, type = uint64())
```

Comment:
`:` creates integers already so you don't need the `L`s:

```
> typeof(1:9)
[1] "integer"
```

Reply:
Fixed in 2d0777eba0628b8d4286d3125ce4ef5a2504141b